[B] Corrected solid property of various materials. Fixes BUKKIT-5671, BUKKIT-5559

JeromSar opened the pull request:
**The Issue:**
Various blocks are marked as solid blocks whilst the player may pass freely through them. Likewise, `FLOWER_POT` was not recognized as a solid block.

**Justification for this PR:**
Previously, `WALL_SIGN` and `SIGN_POST` were recognised as a solid blocks although players can pass through them.

~~Likewise, `WOOD_PLATE`, `STONE_PLATE`, `GOLD_PLATE` and `IRON_PLATE` were treated as solid blocks although none of those blocks prevent the player from passing through.~~ _NMS treats plates as solid blocks. These have been re-added._

Lastly, `FLOWER_POT` was not recognised as a solid block. This block does prevent the player from passing through.

**PR Breakdown:**
This PR corrects the `isSolid()` method in Material.java, removing all materials mentioned above and adding `FLOWER_POT` as a recognised solid block.

**Testing Results and Materials:**
The changes in this PR are too trivial to require individual testing. The changes in this PR compile without compile warnings nor compile errors.

**Relevant PR(s):**
Bukkit/Bukkit#1087 - The original code was inspired from @xa112. His PR did not qualify to the contributing guidelines.

**JIRA Tickets:**
https://bukkit.atlassian.net/browse/BUKKIT-5671
https://bukkit.atlassian.net/browse/BUKKIT-5559

JeromSar committed:
Corrected solid property of various materials
Previously, WALL_SIGN was recognised as a solid block although players can pass through this block.
Likewise, WOOD_PLATE, STONE_PLATE, GOLD_PLATE and IRON_PLATE were treated as solid blocks although none of those blocks prevent the player from passing through.
Lastly, FLOWER_POT was not recognised as a solid block. This block does prevent the player from passing through.

Fixes BUKKIT-5671, BUKKIT-5559

JeromSar committed:
Remove SIGN_POST from recognised solid blocks

feildmaster commented:
We have a test for this... if this is supposed to be the result, could you verify CraftBukkit compiles with this test:
https://github.com/Bukkit/CraftBukkit/blob/master/src/test/java/org/bukkit/PerMaterialTest.java

JeromSar committed:
Re-added plates to solid blocks
Apparantly, NMS treats STONE_PLATE, WOOD_PLATE, IRON_PLATE and GOLD_PLATE as solid blocks. This is odd behaviour since players are able to freely move on them. However, Bukkit must comply with its behaviour and thus these materials have been re-added.

JeromSar commented:
@feildmaster NMS apparently treats plates as solid blocks despite players being able to freely move through them. I've re-addded the plates and CraftBukkit now compiles without test failures.

PerMaterialTest:
![f3409cdae27765d4f4b94b1491f93bc9](https://cloud.githubusercontent.com/assets/1307863/3758381/61ffdb5a-1847-11e4-8317-52a991792c1b.png)

turt2live commented on src/main/java/org/bukkit/Material.java:
```diff
@@ -777,6 +771,11 @@ public boolean isSolid() {
             case DARK_OAK_STAIRS:
             case PACKED_ICE:
+            case FLOWER_POT:
+            case STONE_PLATE:
```
Please correct the unnecessary moving of items

feildmaster commented:
Now the question is, why did our test let craftbukkit compile before with these materials incorrectly set.

(It's not a very good test if it passed before)

JeromSar commented:
@feildmaster It seems all three materials (`FLOWER_POT`, `WALL_SIGN` and `SIGN_POST`) are listed within `INVALIDATED_MATERIALS` in _AbstractTestingBase.java_. They are thus excluded from _PerMaterialTest_.

In fact, a lot more materials are excluded from PerMaterialTest. See [this pastebin](http://pastebin.com/3d9tEpHQ).

These materials are excluded from various test like due to the fact that most of these are magic materials such as `PISTON_MOVING_PIECE` and `DIODE_BLOCK_OFF`).

JeromSar committed:
Move plates back to their original position

feildmaster commented:
Alright... I remember that ugly thing now.

Wish we could have a test which worked with everything.. oh well, chalk it up to the "TODO list"...

feildmaster commented:
> NMS apparently treats plates as solid blocks despite players being able to freely move through them.

Actually, you don't move through plates... you walk on them. ;)
